src/LinkDotNet.NCronJob/Execution/JobExecutor.cs

```diff
@@ -41,7 +45,7 @@ public void CancelJobs()
         }
     }
 
-    public async Task RunJob(RegistryEntry run, CancellationToken stoppingToken)
+    public async Task RunJob(JobDefinition run, CancellationToken stoppingToken)
```

Comment:
I think this may be a good opportunity to separate out the Job definition from the "Run". This here needs to be the Job Run, which has its own Job Run Id that lives only within the lifetime of the Run/Execution. This job run id is different from the Job's Id so that we can correctly correlate the events and logs. We can do this in the state-event-system branch but it will have to be done at some point.

Additionally, chaining the Jobs may also require a top level _Flow Id_ so that the entire job flow can be traced and observed instead of inferring the flow from the parent Job Run.